feat(new-plugin): support Btcfun

Nevermore-Ray opened the pull request:
# BTC.Fun Plugin for Eliza

## Description
BTC.Fun enables users to effortlessly launch and trade Runes/BRC20 tokens on the Bitcoin network in a permissionless environment. Through the platform, anyone can create their own Runes/BRC20 tokens and initiate a minting process. If the funding goal is met, all minting fees are pooled into the Liquidity Pool to boost trading activity. If the goal is not reached, users will receive a 95% refund on minting fees. This functionality is powered by an Eliza plugin, ensuring seamless and efficient operations throughout the process.

## Features
- **Bitcoin Layer1 Token Creation**: Users can create tokens directly on Bitcoin Layer1.
- **BTCFun Functionality**: Enable users to interact with BTCFun features through Merlin.
- **Seamless Integration**: Connect Merlin with the Eliza agent for a smooth user experience.

## Relates To
- Add btcFun plugin into Eliza system.

## Risks
- **Low Risk**: The main implementation risk is that the plugin may not work correctly.

## Background

### What does this PR do?
This PR adds support for Eliza to interact with btcFun, enabling users to:
- Create tokens on Bitcoin Layer1.
- Access BTCFun functionalities through the Eliza agent.

### What kind of change is this?
- **Feature**: Non-breaking change that adds new functionality.

### Documentation Changes
Documentation has been added with configuration details in the plugin's README.md file.

## Testing

### Where should a reviewer start?
- Spin up an agent with the btcFun plugin.
- Follow the testing steps outlined in the README.md file of this plugin.

### Detailed Testing Steps
Please refer to the README.md file in this plugin for detailed testing instructions.

wtfsayo reviewed (changes requested):
Please make sure CI works when plugin is initialised!

`pnpm run integrationTests`
`pnpm run smokeTests`

Nevermore-Ray commented:
> Please make sure CI works when plugin is initialised!
> 
> `pnpm run integrationTests` `pnpm run smokeTests`

Big brother,I've create this pr only commit text "are you ok", but ci still not work.

wtfsayo commented:
will come back here when ci is fixed!

Nevermore-Ray reviewed (commented):
ok

Nevermore-Ray commented:
> will come back here when ci is fixed!

When can we expect the CI to be fixed? Is it possible to merge this PR first since the plugins are relatively independent?

Commit:
read modelConfig from character file

avaer committed:
Merge branch 'develop' into dynamic-plugins2

avaer committed:
Add missing names in plugins

avaer committed:
Remove package.json

avaer committed:
Remove duplicate in package.json

avaer committed:
Update conflicting param name

avaer committed:
Update client-direct build

avaer committed:
Build plugin-bootstrap

avaer committed:
Update types

avaer committed:
Remove plugin di

avaer committed:
Comment out on chain json

tcm390 committed:
fix install

tcm390 committed:
Merge pull request #3261 from elizaOS/tcm-fix-install

fix: pnpm install

tcm390 committed:
fix http dynamic require issue

tcm390 committed:
Merge branch 'develop' into tcm/fix-plugin-ton

tcm390 committed:
Merge pull request #3262 from elizaOS/tcm/fix-plugin-ton

fix: plugin-ton Dynamic require of "http" is not supported

esen committed:
Fix PG query: Use only 'text' subfield name, field name is not needed here

wtfsayo committed:
Update runtime.ts

wtfsayo committed:
fix env name mismatch

nilaysarma committed:
add github issues link

johntad110 committed:
Fix broken links on contributing.md

rebustron committed:
Update broken link README.md

lalalune committed:
fix develop build issue

lalalune committed:
Merge branch 'develop' of http://github.com/ai16z/eliza into HEAD

lalalune committed:
Add chracters

lalalune committed:
remove minified code block

lalalune committed:
Merge pull request #3269 from johntad110/fix-broken-links

Fix broken links on contributing.md

lalalune committed:
Merge pull request #3268 from nilaysarma/add-link-to-issues

docs: Add GitHub issues link to CONTRIBUTING.md file

lalalune committed:
Merge pull request #3264 from esen/fix_pg_memory_query

Fix PG query: Use only 'text' subfield name, field name is not needed here

lalalune committed:
Merge branch 'develop' into tcm-handle-invalid-json

lalalune committed:
Merge pull request #3258 from elizaOS/tcm-handle-invalid-json

fix: handle invalid json

rebustron committed:
typo corr README.md

lalalune committed:
Merge branch 'develop' of http://github.com/ai16z/eliza into HEAD

rebustron committed:
typo corr 2025-01-17.md

reject-i committed:
Update chat_2024-11-19.md

reject-i committed:
Update chat_2024-11-22.md

reject-i committed:
Update chat_2024-06-20.md

reject-i committed:
Update README.md

wtfsayo committed:
refactor runtime

wtfsayo committed:
biome fixes

abcfy2 committed:
fix: handleGoogle(options) lost apiKey

tcm390 committed:
add support for Twitter cookies from environment settings

tcm390 committed:
add support for Twitter cookies from environment settings

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #3270 from rebustron/main

docs: fix typos and update broken link in docs

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #3271 from XxAlex74xX/main

chore: fix spelling issues

btspoony committed:
fix(core): improve OpenAI-like provider endpoint resolution in `generation.ts`

tcm390 committed:
rename variable

tcm390 committed:
Merge branch 'develop' into tcm-fix-DenyLoginSubtask

shakkernerd committed:
chore: add description text to env example

shakkernerd committed:
Merge pull request #3278 from elizaOS/tcm-fix-DenyLoginSubtask

fix: DenyLoginSubtask

azep-ninja committed:
Merge branch 'develop' into fix-rag-optimizations

azep-ninja committed:
optimize template/add suppress init msg

azep-ninja committed:
add actions suppress action ability.

azep-ninja committed:
add actions suppress action ability.

azep-ninja committed:
add actions suppress action ability.

shakkernerd committed:
Merge pull request #3286 from azep-ninja/fix-twitter-suppress-message-add

fix: twitter - add actions suppress action ability.

shakkernerd committed:
Merge branch 'develop' into fix-rag-optimizations

shakkernerd committed:
Merge pull request #3248 from azep-ninja/fix-rag-optimizations

fix: rag optimizations/fixes for context.

shakkernerd committed:
Merge branch 'develop' into fix-optimize-quick-intel-plugin

shakkernerd committed:
Merge pull request #3283 from azep-ninja/fix-optimize-quick-intel-plugin

fix: quick-intel plugin: optimize template/add suppress init msg

shakkernerd committed:
Merge branch 'develop' into fix-discord-suppress-message-add

shakkernerd committed:
Merge pull request #3284 from azep-ninja/fix-discord-suppress-message-add

fix: discord - add actions suppress action ability.

shakkernerd committed:
Merge branch 'develop' into fix-telegram-suppress-message-add

shakkernerd committed:
Merge pull request #3285 from azep-ninja/fix-telegram-suppress-message-add

fix: telegram- add actions suppress action ability.

shakkernerd committed:
Merge branch 'develop' into develop

shakkernerd committed:
Merge pull request #3238 from elpulpo0/develop

feat: (plugin-multiversx) Allow the use of herotag

odilitime committed:
Merge branch 'develop' of https://github.com/elizaos/eliza into odi-fix

odilitime committed:
bump lock

odilitime committed:
Merge branch 'odi-fix' of https://github.com/elizaos/eliza into odi-fix

odilitime committed:
add zod globally

avaer committed:
Move adapter-redis -> cache-redis

odilitime committed:
remove zod

odilitime committed:
bump lockfile

tcm390 committed:
clean json before normalized

odilitime committed:
Merge pull request #3220 from elizaOS/odi-fix

fix: fix docker & types issue

odilitime committed:
bump version to 0.25.6-alpha.1

odilitime committed:
bump lock

avaer committed:
Refactor constructor parameter name in WhatsAppPlugin class

avaer committed:
Internalize adapter-mongodb initialization

avaer committed:
Internalize adapter-pglite initialization

avaer committed:
Internalize adapter-postgres initialization

avaer committed:
Internalize adapter-qdrant initialization

avaer committed:
Internalize adapter-sqlite initialization

avaer committed:
Internalize adapter-supabase initialization

avaer committed:
Update adapter plugin names

avaer committed:
Port alexa plugin

tcm390 committed:
Merge branch 'develop' into tcm-clean-json

avaer committed:
Update client-auto interface

avaer committed:
Update client-deva interface

avaer committed:
Update client-direct interdace

avaer committed:
Update client-discord interface

avaer committed:
Update client-eliza-home interface

avaer committed:
Update client-farcaster interface

avaer committed:
Update client-discord interface

avaer committed:
Update cache-redis name

avaer committed:
Update client-instagram interface

avaer committed:
Update client-lens interface

avaer committed:
Update client-simsai interface

avaer committed:
Update client-slack interface

avaer committed:
Update client-telegram-account interface

avaer committed:
Update client-telegram interface

avaer committed:
Update client-twitter interface

tcm390 committed:
Merge branch 'develop' into tbh/fix-openai-like-provider-endpoint

avaer committed:
Major refactor to xmtp plugin

avaer committed:
Do not check in client-direct dist

odilitime committed:
Merge branch 'develop' of https://github.com/elizaos/eliza into odi-fix

tcm390 committed:
Merge pull request #3281 from btspoony/tbh/fix-openai-like-provider-endpoint

fix(core): improve OpenAI-like provider endpoint resolution in `generation.ts`

avaer committed:
Update client plugins names

avaer committed:
Remove dead dep from client-discord

tcm390 committed:
Merge branch 'develop' into tcm-clean-json

avaer committed:
Remove clients from defaultCharacter.ts

avaer committed:
Remove clients from CharacterSchema

avaer committed:
Defer loading database/cache

avaer committed:
Add plugin adapters tracking

avaer committed:
Update clients tracking in types

avaer committed:
Remove dupe dep

avaer committed:
Remove twitter client dep from bittensor

avaer committed:
Update plugin-echochambers

odilitime committed:
Merge branch 'develop' into odi-fix

avaer committed:
Remove dead dep

avaer committed:
Update agent deps

avaer committed:
Dynamic adapter loading in agent

avaer committed:
Install adapter-sqlite by default

avaer committed:
Rename client-direct

avaer committed:
Update plugin-rabbi-trader deps

avaer committed:
Update lock

odilitime committed:
Merge pull request #3306 from elizaOS/odi-fix

chore: bump version to 0.25.6-alpha.1

tcm390 committed:
Merge branch 'develop' into tcm-clean-json

odilitime committed:
Merge pull request #3307 from elizaOS/develop

chore: develop => main

odilitime committed:
Merge pull request #3301 from elizaOS/tcm-clean-json

fix: clean json before normalized

oxy-Op committed:
Fix plugin-solana-v2 package.json (#3308)

Expected double-quoted property name in JSON at position 825 while parsing '{    "name": "@elizaos/plugin-solana-v2' in /home/oxy/everreach-ai-agent/packages/plugin-solana-v2/package.json

Co-authored-by: Odilitime <janesmith@airmail.cc>

Nevermore-Ray commented:
@shakkernerd  could you help to review and merge this PR

Nevermore-Ray commented:
@tcm390 @odilitime could you help to review this pr，thank you

tcm390 committed:
set initial value to null

tcm390 committed:
return string

tcm390 committed:
return null of empty object

tcm390 committed:
Merge pull request #3311 from elizaOS/tcm-parsing-unit-test

fix: parsing unit test failure

tcm390 committed:
use latest getModelSettings function

tcm390 committed:
fix model test

tcm390 committed:
update LIVEPEER endpoint

tcm390 committed:
Merge branch 'develop' into tcm/fix-models-unit-test

tcm390 committed:
Merge pull request #3312 from elizaOS/tcm/fix-models-unit-test

fix: models unit test

tcm390 commented:
> @tcm390 @odilitime could you help to review this pr，thank you

Looks fine to me :) but could you resolve the conflict?

wtfsayo committed:
wip: save

kesar committed:
allow plugins to interact w messangerManager to post on tg

allow plugins to interact w messangerManager to post on tg

kesar committed:
Merge branch 'develop' into patch-4

wtfsayo committed:
minor change

Bilogweb3 committed:
Update eliza-in-tee.md

Bilogweb3 committed:
Update evaluators.md

Bilogweb3 committed:
Update agent.md

tcm390 committed:
Merge pull request #3313 from TbLtzk/fix/model-config

read modelConfig from character file

tcm390 committed:
Merge branch 'develop' into main

tcm390 committed:
Merge pull request #3317 from Bilogweb3/main

docs: fix typos and update functions

tcm390 committed:
Merge branch 'develop' into main

tcm390 committed:
Merge pull request #3274 from abcfy2/main

fix: handleGoogle(options) lost apiKey

wtfsayo committed:
better runtime logging + refactor + new env norms

wtfsayo committed:
success!!! with direct message!

wtfsayo committed:
Update generation.ts

Commit:
fix conflict

wtfsayo committed:
newer tests

kesar committed:
Merge branch 'develop' into patch-4

shakkernerd committed:
Merge pull request #3314 from kesar/patch-4

feat: allow plugins to interact w messangerManager to post on telegram

aso20455 committed:
Update plugins.md

shakkernerd committed:
chore: add missing version to package.json

shakkernerd committed:
Merge pull request #3325 from elizaOS/fix/packages_json

chore: add missing version property to package.json

shakkernerd committed:
chore: remove remnant files/folders

shakkernerd committed:
Merge pull request #3326 from elizaOS/fix/packages

chore: remove remnant files/folders

vidvidvid committed:
Update post.ts

shakkernerd committed:
Merge branch 'develop' into patch-1

shakkernerd committed:
Merge pull request #3324 from aso20455/patch-1

docs: fix typos in plugins.md

shakkernerd committed:
chore: pnpm lock file

shakkernerd committed:
feat: set package publish access to public

shakkernerd committed:
Merge pull request #3330 from elizaOS/feat/packages

feat: set package publish access to public

shakkernerd committed:
Merge branch 'develop' into vidvidvid-fix/twitter-post-error

shakkernerd committed:
Merge pull request #3327 from vidvidvid/vidvidvid-fix/twitter-post-error

fix: Twitter logging bug

lalalune committed:
pretty

lalalune committed:
pretty

odilitime committed:
Merge pull request #3332 from elizaOS/develop

chore: develop => main

lalalune committed:
Merge branch 'develop' of http://github.com/ai16z/eliza into HEAD

lalalune committed:
Remove redis cache from repo and update

lalalune committed:
remove unncessary depen
dencies

lalalune committed:
move default character to agent

lalalune committed:
Merge pull request #3339 from elizaOS/dynamic-plugins2-merged

Dynamic Plugin Loading (merged_)

lalalune committed:
Merge pull request #3343 from elizaOS/shaw/move-default-character

Shaw/move default character

lalalune committed:
remove verifiable inference concept, will be plugin loaded

lalalune committed:
fix missing verifiableInference

lalalune committed:
fix defualt character

lalalune committed:
Merge pull request #3344 from elizaOS/shaw/remove-verifiable-inference

feat: remove verifiable inference concept, will be plugin loaded

lalalune committed:
Merge pull request #3345 from elizaOS/fix-default-character

fix: fix default character

avaer committed:
Remove commented plugin imports from agent source

avaer committed:
Remove more dead imports

lalalune committed:
fix some testing stuff

lalalune committed:
Merge branch 'develop' into rm-plugins-from-agent

lalalune committed:
Merge pull request #3346 from elizaOS/rm-plugins-from-agent

Remove plugin imports from agent

Commit:
fix conflict

Nevermore-Ray commented:
@tcm390  conflict fixed

lalalune committed:
Merge branch 'main' of http://github.com/ai16z/eliza into HEAD

lalalune committed:
fix ws

lalalune committed:
Merge branch 'develop' of http://github.com/ai16z/eliza into HEAD

lalalune committed:
update runtime to handle ensureUserExists in edge cases

lalalune committed:
Merge branch 'improvement/biome-core-1' of https://github.com/0xSero/eliza into HEAD

Nevermore-Ray committed:
Merge branch 'develop' into btcfun-dev

Nevermore-Ray commented:
@tcm390 Our develop branch is updated too frequently, and there are frequent conflicts. We still hope to merge as soon as possible, as we've already resolved conflicts many times. It's good now. Thanks

wtfsayo committed:
update; test enum stuff

wtfsayo committed:
test more generateSHouldRespond

wtfsayo committed:
fix response

wtfsayo committed:
remove unncessary env validation; DRY Stuff

wtfsayo committed:
Merge remote-tracking branch 'origin/develop' into v1.51

wtfsayo committed:
build and lint fixes

avaer committed:
Readd missing @elizaos-plugins/adapter-sqlite

wtfsayo committed:
rolllll

lalalune committed:
Merge pull request #3357 from elizaOS/sqlite-plugin-missing

Add adapter-sqlite to deps

wtfsayo committed:
Merge branch 'develop' into v1.51

lalalune committed:
Merge branch 'shaw/replace-pnpm-with-bun' of http://github.com/ai16z/eliza into HEAD

lalalune committed:
delete a bunch of crap

lalalune committed:
Merge branch 'develop' of http://github.com/ai16z/eliza into HEAD

lalalune committed:
bun stuff

lalalune committed:
remove crap and lint

lalalune committed:
if someone wants github they can add it back

lalalune committed:
fix ci

lalalune committed:
remove test

lalalune committed:
consolidate client-direct into agent and move sqlite adapter to plugin/sqlite

lalalune committed:
add cli

lalalune committed:
delete crap

lalalune committed:
move i18n, fix build things

lalalune committed:
remove clean from docs, fix smoke test

lalalune committed:
remove gitpod from readme

tcm390 committed:
add anthropic

wtfsayo committed:
Merge remote-tracking branch 'origin/develop' into v1.51

wtfsayo committed:
temp: remove docs package

wtfsayo committed:
re-add docs

tcm390 committed:
handle anthropic provider

tcm390 committed:
make model provider optional

tcm390 committed:
clean code

wtfsayo committed:
remove commented out stuff

tcm390 committed:
rename

tcm390 committed:
use include to check if anthropic function

tcm390 committed:
update provider type

wtfsayo committed:
restore agent/src/index

tcm390 committed:
handle image model

tcm390 committed:
Prevent image generation with unsupported providers for now

tcm390 committed:
clean code

tcm390 committed:
Merge branch 'v1.51' into tcm/support-anthropic

wtfsayo committed:
Update generation.ts

tcm390 committed:
Merge pull request #3364 from elizaOS/tcm/support-anthropic

support anthropic provider

tcm390 committed:
resolve merge conflict

tcm390 committed:
Merge pull request #3367 from elizaOS/tcm/merge-conflict

resolve merge conflict

tcm390 committed:
clean code

lalalune committed:
cli shouldnt start with other stuff

lalalune committed:
Merge branch 'v1.51' of http://github.com/ai16z/eliza into HEAD

lalalune committed:
fix some bugs but providers cooked

HashWarlock committed:
fix build in core to target node23 (#3371)

Nevermore-Ray commented:
@tcm390 If I keep resolving conflicts but can't merge each time, I'll just be resolving conflicts endlessly. This PR has already been going on for half a month. If we're not going to proceed, could you please give a clear response so we don't waste each other's time?

Commit:
fix conflict

tcm390 commented:
Hello,

We are changing our plugin development strategy to be more scalable. We have moved the plugins out into their own repos and we're looking for people to either maintain those or own them on their own Github.

Please make your code changes to this repo: https://github.com/elizaos-plugins/<plugin-name>

If you'd like to be a maintainer,  file an issue in the plugin repo and join our Discord https://discord.gg/elizaos to coordinate.

If you'd like to control the plugin on your own Github, please add an issue to the plugin repo pointing to your repo, and add a modification to the registry. Submit a PR to edit the registry here: https://github.com/elizaos-plugins/registry

Closing this PR for now. Let us know if you have any questions.

Nevermore-Ray commented:
> Hello,
> 
> We are changing our plugin development strategy to be more scalable. We have moved the plugins out into their own repos and we're looking for people to either maintain those or own them on their own Github.
> 
> Please make your code changes to this repo: https://github.com/elizaos-plugins/
> 
> If you'd like to be a maintainer, file an issue in the plugin repo and join our Discord https://discord.gg/elizaos to coordinate.
> 
> If you'd like to control the plugin on your own Github, please add an issue to the plugin repo pointing to your repo, and add a modification to the registry. Submit a PR to edit the registry here: https://github.com/elizaos-plugins/registry
> 
> Closing this PR for now. Let us know if you have any questions.

Done. https://github.com/elizaos-plugins/registry/issues/24
